docs/sources/source/tutorials/old_callbacks.rst

```diff
@@ -0,0 +1,206 @@
+
+    nf.train(
+        callbacks=[loggercallback, savercallback, evalcallback],
+        ...)
+
+SimpleLossLoggerCallback
+------------------------
+SimpleLossLoggerCallback is used to
+log training metrics such as loss and time per step to screen and to
+tensorboard. SimpleLossLoggerCallback has one required arguments, and two
+arguments that we recommend to overwrite. It requires tensors which is a list
+of NMTensors that print_func(), get_tb_values(), and log_to_tb_func will
+receive as input during
+training. The two reccomended arguments to override are print_func(), and
+either get_tb_values() or log_to_tb_func().
+
+print_func() should be used to log values to screen. We recommend using
+logging.info() in place
+of print(). For example, it can be used to print the loss value:
+
+.. code-block:: python
+
+    from nemo import logging
+    def my_print_func(tensors):
+        logging.info(f"Loss {tensors[0]}")
+
+We provide two methods to log to tensorboard: get_tb_values() and
+log_to_tb_func(). For simple use case of logging scalars, we recommend
+get_tb_values(). For advanced use cases such as pictures or audio, we
+recommend log_to_tb_func().
+
+get_tb_values() is used to return values to be logged to tensorboard. It should
+return a list of 2-element tuples, where the first element is a string
+representing the tensorboard tag, and the second element is the scalar value to
+log. Note we currently only support scalar values. Note to use get_tb_values(),
+tb_writer should also be defined.
+
+.. code-block:: python
+
+    def my_get_tb_values(tensors):
+        return [("Train_Loss", tensors[0])]
+
+log_to_tb_func() takes three arguments: the
+`tensorboardX.SummaryWriter <https://tensorboardx.readthedocs.io/en/latest/tensorboard.html>`_
+, a list of evaluated tensors, and the current step. The user can then use the
+SummaryWriter class to add images, audio, and more. For example:
+
+.. code-block:: python
+
+    def log_to_tb_func(swriter, tensors, step):
+        swriter.add_scalar("Train_Loss", tensors[0], step)
+        swriter.add_audio("Train_Sample", tensors[1][0], step)
+
+SimpleLossLoggerCallback can be constructed as follows:
+
+.. code-block:: python
+
+    from functools import partial
+
+    loggercallback = nemo.core.SimpleLossLoggerCallback(
+        # Define tensors that we want to pass to print_func, and get_tb_values
+        tensors=[train_loss],
+        # Pass the print function that we want to use
+        print_func=my_print_func,
+        # Pass the function that returns tensorboard tags and scalars
+        get_tb_values=my_get_tb_values,
+        # How often we want to call this callback
+        step_freq=500,
+        # The tensorboard writer object we want to use, it should be
+        # automatically created by neural_factory if create_tb_writer was
+        # set to True during neural_factory construction
+        tb_writer=neural_factory.tb_writer)
+    )
```

Comment:
This is great, I remember was impressed by callback description in PyTorchLightning, nice!🥇

Reply:
wait... I didn't get that, I thought this is the new one... my bad :]